Add item requests and gateway

7Jenko opened the pull request.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Добавить запрос вещи.
Разбить приложение ShareIt на два — shareIt-server и shareIt-gateway.
Добавить тестирование.

7Jenko committed:
Fix tests.

7Jenko committed:
Fix tests.

7Jenko committed:
Fix tests.

7Jenko committed:
Fix tests.

7Jenko committed:
Fix tests.

slavrikk commented on gateway/pom.xml:
```diff
@@ -0,0 +1,287 @@
+
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
```
Для gateway зависимость spring data jpa излишняя

slavrikk commented on gateway/pom.xml:
```diff
@@ -0,0 +1,287 @@
+        </dependency>
+
+        <dependency>
+            <groupId>org.postgresql</groupId>
```
Базы данных в gateway нет, эта зависимость излишняя

slavrikk commented on gateway/pom.xml:
```diff
@@ -0,0 +1,287 @@
+        </dependency>
+
+        <dependency>
+            <groupId>com.h2database</groupId>
```
Базы данных в gateway нет, эта зависимость излишняя

slavrikk commented on gateway/src/main/java/exceptions/ErrorHandler.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    @ExceptionHandler(InternalServerErrorException.class)
+    public ResponseEntity<Map<String, String>> handleInternalServerErrorExceptions(Exception e) {
```
Этот хендлер кажется лишний, пусть непредвиденные ошибки падают в `handleGeneralExceptions`

slavrikk commented on gateway/src/main/java/ru/practicum/shareit/booking/controller/BookingController.java:
```diff
@@ -0,0 +1,70 @@
+			Object response = bookingClient.createBooking(userId, bookingRequestDto);
+			return ResponseEntity.ok(response);
+		} catch (NotFoundException e) {
+			log.error("Item not found: {}", e.getMessage());
```
Для перехвата ошибок есть хендлеры

slavrikk commented on server/src/main/java/ru/practicum/shareit/booking/dto/BookingRequestDto.java:
```diff
@@ -0,0 +1,19 @@
+@AllArgsConstructor
+@NoArgsConstructor
+public class BookingRequestDto {
+    @NotNull
```
Для всех dto запросов в модуле server следует убрать аннотации валидации, так как все запросу валидируются на уровне gateway

slavrikk replied:
Замечание для всего кода

7Jenko committed:
Убрать лишние зависимости в gateway/pom.xml.
Убрать лишний Handler handleInternalServerErrorExceptions.
Заменить блок try-catch на Handler BookingController.
Убрать аннотации валидации для всех dto запросов в модуле server.
